cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/workflow/DefaultWorkflowConditionConfigurer.java

```diff
@@ -76,8 +78,8 @@ public DefaultWorkflowConditionConfigurer(T parentConfigurer, String predicateCl
   @SuppressWarnings("unchecked")
   public WorkflowConditionConfigurer<? extends WorkflowConditionConfigurer<T>> condition(
     Predicate<WorkflowContext> predicate) {
-    String predicateClassName = predicate.getClass().getName();
-    return new DefaultWorkflowConditionConfigurer<>(this, predicateClassName);
+    return new DefaultWorkflowConditionConfigurer<>(predicate.getClass().getSimpleName(), this,
```

Comment:
While assigning the node Ids to the fork nodes which are inside the condition node branches, we recreate the condition nodes in the `createConditionNodeWithId` method. At this point we do not have access to the predicate. So I think its better to pass the simple name and name of the predicate class instead of passing the predicate.